pyat/at/lattice/utils.py

```diff
@@ -434,42 +573,47 @@ def checkname(pattern: str) -> ElementFilter:
     # noinspection PyUnresolvedReferences
     r"""Checks the name of an element
 
-        Returns a function to be used as an :py:class:`.Element` filter,
-        which checks the name of the provided :py:class:`.Element`.
-        This function can be used to extract from a ring all elements
-        having a given name.
+    Returns a function to be used as an :py:class:`.Element` filter,
+    which checks the name of the provided :py:class:`.Element`.
+    This function can be used to extract from a ring all elements
+    having a given name.
 
-        Parameters:
-            pattern: Desired :py:class:`.Element` name. Unix shell-style
-              wildcards are supported (see :py:func:`fnmatch.fnmatch`)
+    Parameters:
+        pattern: Desired :py:class:`.Element` name. Unix shell-style
+          wildcards are supported (see :py:func:`fnmatch.fnmatch`)
 
-        Returns:
-            checkfun (ElementFilter):   Element filter function
+    Returns:
+        checkfun (ElementFilter):   Element filter function
 
-        Examples:
+    Examples:
 
-            >>> qps = filter(checkname('QF*'), ring)
+        >>> qps = filter(checkname('QF*'), ring)
 
-            Returns an iterator over all with name starting with ``QF``.
-        """
+        Returns an iterator over all with name starting with ``QF``.
+    """
     return lambda el: fnmatch(el.FamName, pattern)
 
 
 # noinspection PyIncorrectDocstring
-def get_cells(ring: Sequence[Element], *args) -> BoolRefpts:
-    # noinspection PyUnresolvedReferences
+def get_cells(ring: Sequence[Element], refpts: Refpts, *args) -> BoolRefpts:
+    # noinspection PyShadowingNames
     r"""
     get_cells(ring, filtfunc) -> BoolRefpts
+    get_cells(ring, element_type) -> BoolRefpts
     get_cells(ring, attrname) -> BoolRefpts
     get_cells(ring, attrname, attrvalue) -> BoolRefpts
     Returns a bool array of element indices, selecting ring elements.
 
+    With minor modifications, this function can be replaced by
```

Comment:
What are you trying to indicate with this comment? You have already mentioned elsewhere that this does not have the same behaviour as Lattice.bool_refpts.

Are we to treat one of these as deprecated?

Reply:
> Are we to treat one of these as deprecated?

I would say so (see below) …